lib/cli/do_nfs_volumes.js

```diff
@@ -0,0 +1,341 @@
+            }
+
+            assert.arrayOfObject(ctx.imagesVersions, 'ctx.imagesVersions');
+
+            self.progress('Checking dependencies are up to date');
+
+            outdatedVersions =
+                ctx.imagesVersions.filter(function filterOutdated(versionInfo) {
+                    var serviceName = versionInfo.serviceName;
+                    var version = versionInfo.version;
+
+                    assert.string(serviceName, 'serviceName');
+                    assert.string(version, 'version');
+
+                    self.progress('Checking service ' + serviceName + ' at ' +
+                        'version ' + version + ' is up to date');
+
+                    /*
+                     * When this is submitted to be merged in master, this
+                     * ^tritonnfs pattern needs to be updated to a different
+                     * test that checks that "version" represents a version that
+                     * is at least as recent as the first build of the
+                     * corresponding service with NFS volumes support.
+                     */
+                    if (/^tritonnfs/.test(version)) {
+                        return false;
+                    }
+
+                    return true;
+                });
+
+            if (outdatedVersions && outdatedVersions.length > 0) {
+                err = new Error('Found outdated core services: ' +
+                    outdatedVersions.map(renderVersionInfo).join(', '));
+            }
+
+            function renderVersionInfo(versionInfo) {
+                return versionInfo.serviceName + ' at version ' +
+                    versionInfo.version;
+            }
+
+            next(err);
+        },
+
+        function updateNfSharedVolumesFlag(ctx, next) {
+            var desiredNfsSharedVolumesFlagValue = true;
+            if (opts.disable === true) {
+                desiredNfsSharedVolumesFlagValue = false;
+            }
+
+            self.progress('Checking if experimental_nfs_shared_volumes=' +
+                desiredNfsSharedVolumesFlagValue + ' in SDC app...');
+
+            function _nfsSharedVolumesUpdated(err, result) {
+                var errMsg;
+
+                if (!err) {
+                    if (result === desiredNfsSharedVolumesFlagValue) {
+                        self.progress('experimental_nfs_shared_volumes set ' +
+                            'to ' + desiredNfsSharedVolumesFlagValue + ' on ' +
+                            'SDC app');
+                    } else {
+                        errMsg = 'Could not set ' +
+                            'experimental_nfs_shared_volumes to ' +
+                            desiredNfsSharedVolumesFlagValue + ' on SDC app';
+                        self.progress(errMsg);
+                        err = new Error(errMsg);
+                    }
+                }
+
+                next(err);
+            }
+
+            if (self.sdcadm.sdc.metadata.experimental_nfs_shared_volumes !==
```

Comment:
It will be called by `_nfsSharedVolumesUpdated`. It's not very readable and clear though, so I'll see if I can make that code easier to read.